### ISSUE TITLE
Could not deserialize signed custom property containing dictionary property

### ISSUE THREAD

**Issue by MichalOberreiter** [riganti/dotvvm#1379]
Hi guys,

little trouble with deserialization. When I add dictionary property to any class  used in property with ProtectMode.SignData, it can't be deserialized after postback.
DotVVM.AspNetCore 4.0.3, .NET 6.0

```
public class DefaultViewModel : DotvvmViewModelBase
{
    [Protect(ProtectMode.SignData)]
     public Human Man{ get; set; } = new Human();

    public class Human
   {
        public Dictionary<string, string> Dict { get; set; } = new Dictionary<string, string>();
   }
}
```

```
{
ClassName:"System.Exception"
Message:"TokenType = StartObject was expected."
Data:null
InnerException:null
HelpURL:null
StackTraceString:" at (ArrayClosure , JsonReader , JsonSerializer , Object , EncryptedValuesReader )\r\n at DotVVM.Framework.ViewModel.Serialization.ViewModelJsonConverter.ReadJson(JsonReader reader, Type objectType, Object existingValue, JsonSerializer serializer)\r\n at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.DeserializeConvertable(JsonConverter converter, JsonReader reader, Type objectType, Object existingValue)\r\n at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateList(IList list, JsonReader reader, JsonArrayContract contract, JsonProperty containerProperty, String id)\r\n at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)\r\n at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)\r\n at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)\r\n at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)\r\n at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)\r\n at DotVVM.Framework.ViewModel.Serialization.ViewModelSerializationMap.Deserialize(JsonSerializer serializer, JsonReader reader, ViewModelPropertyMap property, Object existingValue)\r\n at (ArrayClosure , JsonReader , JsonSerializer , Object , EncryptedValuesReader )\r\n at DotVVM.Framework.ViewModel.Serialization.ViewModelJsonConverter.Populate(JsonReader reader, JsonSerializer serializer, Object value)\r\n at DotVVM.Framework.ViewModel.Serialization.DefaultViewModelSerializer.PopulateViewModel(IDotvvmRequestContext context, String serializedPostData)"
RemoteStackTraceString:null
RemoteStackIndex:0
ExceptionMethod:null
HResult:-2146233088
Source:"FastExpressionCompiler"
WatsonBuckets:null
}
```

**Comment by exyi** [comment]
Thanks for reporting it, I found the issue and it's fixed in  #1391. After someones reviews it, do you want me to include it in the 4.0.X bugfix release, or is switching to 4.1 preview fine for you?

**Comment by MichalOberreiter** [comment]
Thanks for fix, I can switch to 4.1, no problem.

________________________________
Od: Stanislav Lukeš ***@***.***>
Odesláno: sobota 21. května 2022 16:28
Komu: riganti/dotvvm ***@***.***>
Kopie: MichalOberreiter ***@***.***>; Author ***@***.***>
Předmět: Re: [riganti/dotvvm] Could not deserialize signed custom property containing dictionary property (Issue #1379)


Thanks for reporting it, I found the issue and it's fixed in #1391<https://github.com/riganti/dotvvm/pull/1391>. After someones reviews it, do you want me to include it in the 4.0.X bugfix release, or is switching to 4.1 preview fine for you?

**Comment by exyi** [comment]
4.1-preview03 should contain the bugfix, let us know if it still does not work
